### ISSUE TITLE
'X' or '' that can be replaced with abap_true or abap_false

### ISSUE THREAD

**Issue by larshp** [larshp/abapOpenChecks#22]

**Comment by DennstedtB** [comment]
will work on this.

**Comment by larshp** [comment]
not it can be difficult to implement, abap_true and abap_false should be used when the type is abap_bool

setting something like `order_type = 'X'.` I would regard as a false positive, as order_type should not contain boolean values

**Comment by DennstedtB** [comment]
Yep. I know that this will be difficult. But this can be done, somehow.

**Comment by larshp** [comment]
yea, sure, just have to build the AST, and defer the types 😄 

I gave up on doing this part in ABAP, since the compiler changes between versions, plus its difficult to do real testing, or alternatively everything must be built from scratch.

Hence the https://github.com/abaplint/abaplint project
